build/functions.sh

```diff
@@ -13,16 +13,44 @@
 #
 # shellcheck shell=ash
 
-if ! type -p realpath; then
-	# by default, macOS does not have realpath
-	realpath() {
-		ls "$(
-			cd "$(dirname "$0")"
-			pwd -P # -P resolves symlinks
-		)/$(basename "$0")"
-	}
-	export -f realpath
-fi;
+# macOS's version of realpath does not resolve symlinks, so we add a function
+# for it.
+get_realpath() {
```

Comment:
<img width="676" alt="image" src="https://github.com/apache/trafficcontrol/assets/29043139/02959ba7-1532-4cea-b6b9-19749d3a9b1d">
even if I have realpath, it doesn't work on MacOS

Reply:
Fixed for macOS and BSDs by checking for  `grealpath` as well in e298e4a4b.